presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoConnection.java

```diff
@@ -162,6 +162,7 @@ public void close()
             throws SQLException
     {
         closed.set(true);
+        queryExecutor.close();
```

Comment:
Query executor can be shared among multiple SQL connections. Don't close it here. It is enough to close it upon the invalidation in cache.